aioamqp/compat.py

```diff
@@ -9,3 +9,17 @@
     from asyncio import ensure_future
 except ImportError:
     ensure_future = asyncio.async
+
+
+def iscoroutinepartial(fn):
+    # http://bugs.python.org/issue23519
+
+    while True:
+        parent = fn
+
+        fn = getattr(parent, 'func', None)
```

Comment:
initializing outside of the loop seems like a good idea and would also avoid using `True`:
```python
parent = fn
while fn is not None:
    parent, fn = fn, getattr(parent, 'func', None)
```